git_test.go

```diff
@@ -39,3 +51,171 @@ func TestCountContributorsFlags(t *testing.T) {
+	defer cleanUpPolicy()
+	_, cleanUpWatch := securityTestUtils.CreateTestProjectKeyWatch(t, policyName, "project-key-jas-violations-watch", *securityTests.JfrogTestProjectKey)
+	defer cleanUpWatch()
+
+	// Run the audit command with git repo and verify violations are reported to the platform.
+	createTestProjectRunGitAuditAndValidate(t,
+		filepath.Join(filepath.FromSlash(securityTests.GetTestResourcesPath()), "git", "projects", "issues"),
+		auditCommandTestParams{Format: string(format.SimpleJson), ProjectKey: *securityTests.JfrogTestProjectKey},
+		xrayVersion, xscVersion, results.NewFailBuildError().Error(),
+		validations.ValidationParams{
+			Total: &validations.TotalCount{Violations: 16},
+			// Check that we have at least one violation for each scan type. (IAC is not supported yet)
+			Violations: &validations.ViolationCount{ValidateScan: &validations.ScanCount{Sca: 1, Sast: 1, Secrets: 1}},
+		},
+	)
+}
+
+func TestXrayAuditJasSkipNotApplicableCvesViolations(t *testing.T) {
+	xrayVersion, xscVersion, testCleanUp := integration.InitGitTest(t, services.MinXrayVersionGitRepoKey)
+	defer testCleanUp()
+
+	projectPath := filepath.Join(filepath.FromSlash(securityTests.GetTestResourcesPath()), "git", "projects", "issues")
+
+	// Create policy and watch for the git repo so we will also get violations - This watch DO NOT skip not-applicable results
+	var firstPolicyCleaned, firstWatchCleaned bool
+	policyName, cleanUpPolicy := securityTestUtils.CreateTestSecurityPolicy(t, "without-skip-non-applicable-policy", utils.Low, false, false)
+	defer func() {
+		if !firstPolicyCleaned {
+			cleanUpPolicy()
+		}
+	}()
+	watchName, cleanUpWatch := securityTestUtils.CreateWatchForTests(t, policyName, "without-skip-not-applicable-watch", xscutils.GetGitRepoUrlKey(validations.TestMockGitInfo.GitRepoHttpsCloneUrl))
+	defer func() {
+		if !firstWatchCleaned {
+			cleanUpWatch()
+		}
+	}()
+
+	// Run the git audit command and verify violations are reported to the platform.
+	createTestProjectRunGitAuditAndValidate(t, projectPath,
+		auditCommandTestParams{Format: string(format.SimpleJson), Watches: []string{watchName}, DisableFailOnFailedBuildFlag: true},
+		xrayVersion, xscVersion, "",
+		validations.ValidationParams{
+			Violations: &validations.ViolationCount{
+				ValidateScan:                &validations.ScanCount{Sca: 8, Sast: 2, Secrets: 2},
+				ValidateApplicabilityStatus: &validations.ApplicabilityStatusCount{NotApplicable: 1, NotCovered: 7},
+			},
+			ExactResultsMatch: true,
+		},
+	)
+
+	// We clean the initially created Policy and Watch that are related to the Git Repo resource, because we must have all related policies with skipNotApplicable=true
+	cleanUpWatch()
+	firstWatchCleaned = true
+	cleanUpPolicy()
+	firstPolicyCleaned = true
+
+	// Create policy and watch for the git repo so we will also get violations - This watch DO NOT skip not-applicable results
```

Comment:
The comment here should be "This watch SKIP not-applicable results"

Reply:
done